src/base/static/components/translator.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import qs from "qs";
+
+import { LanguageConsumer } from "../context/language";
+
+const Translator = props => {
+  const [translatedMsg, setTranslatedMsg] = useState(props.msg);
+  useEffect(
+    () => {
+      if (
+        props.defaultLanguage.code === props.currentLanguage.code ||
+        !GOOGLE_TRANSLATE_API_TOKEN
+      ) {
+        return;
+      }
+
+      async function translate() {
+        // TODO: html format
+        // https://cloud.google.com/translate/docs/languages
+        const response = await fetch(
```

Comment:
We're using a simple `GET` request on the Google Translate API here. I'm not sure this will work for long pieces of text however. There is also a way to make a `POST` request, but we need to provide authorization creds, which I think would mean setting up a small proxy service.

But this might be an ideal use case for AWS Lambda.

Reply:
From what I can tell we would need to use OAuth, which requires a JSON file of credentials and other info. I don't think this file should be exposed to a browser, or at least I haven't found any examples of it being used that way.

Here's the overview from Google's documentation: https://cloud.google.com/docs/authentication/getting-started